string text
float64 speed_scale